backend/tests/data_products_datasets/test_router.py

```diff
@@ -26,6 +28,24 @@ def test_request_data_product_link(
         )
         assert link.status_code == 200
 
+    def test_request_data_product_link_by_admin(
```

Comment:
I feel like this tests the wrong thing, because of the admin dataset and admin product, but I might be wrong.
I would test that the admin user can link any product and dataset together, regardless of ownership